app/views/shared/_admin_flash.html.erb

```diff
@@ -1,8 +1,12 @@
-<div class="container">
+<div class="container" data-controller="flash">
   <% AdminFlash.showable_messages(disabled_flash_ids_from_cookies).each do |admin_flash|  %>
     <div class="alert alert-success fade show admin-flash">
-      <button class="close "data-expire="<%= admin_flash.expires %>" data-id="<%= admin_flash.id %>" data-dismiss="alert">&times;</button>
+      <button class="close "data-expire="<%= admin_flash.expires %>" data-id="<%= admin_flash.id %>" data-dismiss="alert" data-action="click->flash#setCookie" data-flash-target="close">&times;</button>
```

Comment:
We could use the [stimulus value api](https://stimulus.hotwired.dev/reference/values) with the data attributes here to tidy up the controller a little bit.

```html
data-admin-flash-expires-at-value="<%= admin_flash.expires %>"
data-admin-flash-id-value="<%= admin_flash.id %>"
```

Then in the controller, we won't need the close target since we can get the values directly:
```javascript
export default class AdminFlashController extends Controller {
  static values = {
    expiresAt: String,
    id: Number,
  };

  setCookie() {
    const cookieDate = this.convertToCookieDate();
    document.cookie = `admin_flash_${this.idValue}=disabled; expires=${cookieDate}; path=/`;
  }

 convertToCookieDate() {
    const parts = this.expiresAtValue.split('-');
    ...
  }
}

```

I think the value attributes need to be on the same element where the data-controller attribute is set. We could move the values up to the container or add the `data-controller="admin-flash"` attribute to the button.